bin/main.rb

```diff
@@ -1,3 +1,45 @@
 #!/usr/bin/env ruby
 
-puts "Hello world!"
+puts "Welcome to Tic-tac-toe!"
+
+puts "Input x player name:"
+x_player = gets.chomp
+puts "Welcome #{x_player}"
+
+puts "Input o player name:"
+o_player = gets.chomp
+puts "Welcome #{o_player}"
+
+puts "Let's start"
+
+turn = 0
+winner = false
+while turn < 9
+  begin
+    puts "It's #{current_player}'s turn. Select a square from #{available_squares}"
+    move = gets.chomp
+    add_move(move)
+  rescue => exception
+      puts "Invalid move, try again"
+      retry
+  else
+    turn += 1
+    puts "Your move was added to the board"
+    puts display_board
+  end
```

Comment:
Did you run your code ?
I ran it and I never go into the code from `line 26 to 29`.
```ruby
$ bin/main.rb 
Welcome to Tic-tac-toe!
Input x player name:
Alice
Welcome Alice
Input o player name:
Bob
Welcome Bob
Let's start
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
Invalid move, try again
........
........
Invalid move, try again
Invalid move, try ag^CTraceback (most recent call last):
	4: from bin/main.rb:18:in `<main>'
	3: from bin/main.rb:23:in `rescue in <main>'
	2: from bin/main.rb:23:in `puts'
	1: from bin/main.rb:23:in `puts'
bin/main.rb:23:in `write': Interrupt
```